### PR TITLE
Check the return status of `_flush_memtable_async()`

### PR DESCRIPTION
**Pull request by lingbin**
This commit also contains some adjustments of the forward declaration

### CODE REVIEW

File: be/src/runtime/tablets_channel.h [hunk 1]
```diff
@@ -36,25 +36,17 @@ struct TabletsChannelKey {
...
 
-    TabletsChannelKey(const PUniqueId& pid, int64_t index_id_)
-        : id(pid), index_id(index_id_) { }
+    TabletsChannelKey(const PUniqueId& pid, int64_t idx_id) : id(pid), index_id(idx_id) { }
 
-    ~TabletsChannelKey() noexcept { }
+    ~TabletsChannelKey() { }
 
-    bool operator==(const TabletsChannelKey& rhs) const noexcept {
+    bool operator==(const TabletsChannelKey& rhs) const {
```
Comment (imay) [inline be/src/runtime/tablets_channel.h]: Is there any reason to drop noexcept?

File: be/src/runtime/tablets_channel.cpp [hunk 2]
```diff
@@ -104,20 +102,20 @@ Status TabletsChannel::add_batch(const PTabletWriterAddBatchRequest& params) {
     return Status::OK();
 }
 
-Status TabletsChannel::close(int sender_id, bool* finished,
+Status TabletsChannel::close(int sender_idx, bool* finished,
```
Comment (imay) [inline be/src/runtime/tablets_channel.cpp]: why change sender_id to sender_idx? I think if there is no strong reason, better to keep the original name.

File: be/src/runtime/tablets_channel.cpp [hunk 3]
```diff
@@ -25,11 +25,9 @@
 
 namespace doris {
 
-TabletsChannel::TabletsChannel(
-        const TabletsChannelKey& key,
-        MemTracker* mem_tracker): 
-    _key(key), _closed_senders(64) {
-    _mem_tracker.reset(new MemTracker(-1, "tablets channel", mem_tracker));
+TabletsChannel::TabletsChannel(const TabletsChannelKey& key, MemTracker* parent) :
```
Comment (imay) [inline be/src/runtime/tablets_channel.cpp]: I don't think parent is a good name, parent looks like parent of tabletsChannel not only a MemTracker.

**Review by imay — CHANGES_REQUESTED** [review]
Please keep the name original except there is a strong reason.

**Comment by lingbin** [comment]
> Please keep the name original except there is a strong reason.

All name changes have been removed.

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**